src/ethereum_test_base_types/base_types.py

```diff
@@ -75,6 +75,50 @@ def or_none(cls: Type[N], input: N | NumberConvertible | None) -> N | None:
         return cls(input)
 
 
+class Wei(Number):
+    """
+    Class that helps represent wei that can be parsed from strings
+    """
+
+    def __new__(cls, input: NumberConvertible | N):
+        """
+        Creates a new Number object.
+        """
+        if isinstance(input, str):
+            words = input.split()
+            multiplier = 1
+            assert len(words) <= 2
+            value_str = words[0]
+            if len(words) > 1:
+                unit = words[1]
+                if unit == "wei":
+                    multiplier = 1
+                elif unit == "kwei":
+                    multiplier = 10**3
+                elif unit == "mwei":
+                    multiplier = 10**6
+                elif unit == "gwei":
+                    multiplier = 10**9
+                elif unit == "szabo":
+                    multiplier = 10**12
+                elif unit == "finney":
+                    multiplier = 10**15
+                elif unit == "ether":
+                    multiplier = 10**18
+                else:
+                    raise ValueError(f"Invalid unit {unit}")
```

Comment:
Normalize with `lower()`; allow upper/mixed case strings:
```suggestion
            if len(words) > 1:
                unit = words[1].lower()
                multiplier = cls._get_multiplier(unit)
```